src/os_win/os_mtx_cond.c

```diff
@@ -59,6 +58,23 @@ __wt_cond_wait_signal(
 	EnterCriticalSection(&cond->mtx);
 	locked = true;
 
+	/*
+	 * It's possible to race with threads waking us up. That's not a problem
+	 * if there are multiple wakeups because the next wakeup will get us, or
+	 * if we're only pausing for a short period. It's a problem if there's
+	 * only a single wakeup, our waker is likely waiting for us to exit.
+	 * After acquiring the mutex (so we're guaranteed to be awakened by any
+	 * future wakeup call), optionally check if we're OK to keep running.
+	 * This won't ensure our caller won't just loop and call us again, but
+	 * at least it's not our fault.
+	 *
+	 * Assert we're not waiting longer than a few seconds if not checking
+	 * the run status.
+	 */
+	WT_ASSERT(session, run_func != NULL || usecs < 5 * WT_MILLION);
```

Comment:
I don't believe anything will break because all of our current calls to `__wt_cond_auto_alloc` set a max of 1 second.

That said, I don't see a need to tighten the test down. It's not wrong to create a condition variable with a max-wait of more than 1 second, not specifying a race-check function is only a problem if the wait is long enough to affect shutdown.

If you still think it's better at 1 second, feel free to tighten it down.